var/spack/repos/builtin/packages/py-deephyper/package.py

```diff
@@ -20,22 +20,28 @@ class PyDeephyper(PythonPackage):
     license("BSD-3-Clause")
 
     version("master", branch="master")
+    version("0.6.0", sha256="cda2dd7c74bdca4203d9cd637c4f441595f77bae6d77ef8e4a056b005357de34")
     version("0.4.2", sha256="ee1811a22b08eff3c9098f63fbbb37f7c8703e2f878f2bdf2ec35a978512867f")
 
     depends_on("python@3.7:3.9", type=("build", "run"))
 
+    depends_on("py-setuptools@42:", type="build", when="@0.6.0")
     depends_on("py-setuptools@40:49.1", type="build")
     depends_on("py-wheel@0.36.2", type="build")
-    depends_on("py-cython@0.29.24:2", type="build")
+    depends_on("py-cython@0.29.24:", type="build", when="@0.6.0")
+    depends_on("py-cython@0.29.24:2", type="build", when="@0.4.2")
 
     depends_on("py-configspace@0.4.20:", type=("build", "run"))
     depends_on("py-dm-tree", type=("build", "run"))
-    depends_on("py-jinja2@:3.0", type=("build", "run"))
+    depends_on("py-jinja2@:3.1", type=("build", "run"), when="@0.6.0")
+    depends_on("py-jinja2@:3.0", type=("build", "run"), when="@0.4.2")
+    depends_on("py-numpy@1.20:", type=("build", "run"), when="@0.6.0")
     depends_on("py-numpy", type=("build", "run"))
     depends_on("py-pandas@0.24.2:", type=("build", "run"))
     depends_on("py-packaging", type=("build", "run"))
```

Comment:
According to the pyproject.toml file, there is a specific version required for macOS M1 (below).  Should this be reflected as well?

```
    "packaging==20.5; platform_machine=='arm64'",  # macos M1
```

Reply:
This could probably be less strict now `packaging>=20.5`. If I remember correctly this was at the time of release of the new mac with m1 ship.